src/modules/utils/readpin/ReadPin.cpp

```diff
@@ -0,0 +1,59 @@
+#include "libs/Kernel.h"
+#include "ReadPin.h"
+#include "libs/nuts_bolts.h"
+#include "libs/utils.h"
+#include "Config.h"
+#include "SlowTicker.h"
+#include "libs/SerialMessage.h"
+#include "libs/StreamOutput.h"
+#include "checksumm.h"
+#include "ConfigValue.h"
+#include "StreamOutputPool.h"
+
+using namespace std;
+
+#define gcode_checksum               CHECKSUM("gcode")
+#define pin_checksum                 CHECKSUM("pin")
+
+ReadPin::ReadPin() {
+}
+
+ReadPin::ReadPin(uint16_t name) {
+    name_checksum = name;
+}
+
+void ReadPin::on_module_loaded() {
+    THEKERNEL->slow_ticker->attach(5, this, &ReadPin::pin_tick);
```

Comment:
This is an absolute waste of CPU resources, just read the pin on demand not 5 times a second all the time.